### PR TITLE
feat: replace deprecated nav components

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #1477

refactor navigation components

- `OnyxFlyoutMenu`: rename default slot to options
- previous `OnyxNavItem` component is now `OnyxNavButton` with changed API
- new `OnyxNavItem` is now only intended to be used as children for the new `OnyxNavButton`
- `OnyxUserMenu`: property `avatar` now only accepts a string, not an object
- fix(OnyxUserMenu): use correct font weight for username

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/modules/OnyxNavButton/NavButton-mobile-open-children--chromium-linux.png
Comment (larsrickert) [inline packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/modules/OnyxNavButton/NavButton-mobile-open-children--chromium-linux.png:1]: Add `label` property to OnyxNavItem and implement external icon by default.
Custom slot is still supported

File: packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/modules/OnyxUserMenu/User-menu-chromium-linux.png
Comment (larsrickert) [inline packages/sit-onyx/playwright/snapshots/components/OnyxNavBar/modules/OnyxUserMenu/User-menu-chromium-linux.png:1]: hover should open flyout

File: packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxNavItem/OnyxNavItem.stories.ts [hunk 1]
```diff
@@ -74,13 +55,3 @@ export const WithCustomContent = {
     default: () => ["Custom label", h(OnyxBadge, { dot: true, color: "warning" })],
   },
 } satisfies Story;
-
-/**
- * This example shows the nav item with external link.
- */
-export const WithExternalLink = {
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxNavItem/OnyxNavItem.stories.ts:81]: Add it back when label property is added

File: packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxUserMenu/OnyxUserMenu.stories.ts [hunk 2]
```diff
@@ -40,8 +40,8 @@ export const Default = {
     username: "Jane Doe",
     description: "Company Name",
     default: () => [
-      h(OnyxColorSchemeMenuItem, { modelValue: "auto" }),
-      h(OnyxListItem, { color: "danger" }, () => [h(OnyxIcon, { icon: logout }), "Logout"]),
+      h(OnyxMenuItem, () => [h(OnyxIcon, { icon: settings }), "Settings"]),
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxUserMenu/OnyxUserMenu.stories.ts:43]: Add back OnyxColorSchemeMenuItem

**Review by MajaZarkova — APPROVED** [review]

**Pull request merged**